compiler/rustc_mir/src/transform/coverage/spans.rs

```diff
@@ -158,6 +173,34 @@ impl CoverageSpan {
             .collect::<Vec<_>>()
             .join("\n")
     }
+
+    /// If the span is part of a macro, and the macro is visible (expands directly to the given
+    /// body_span), returns the macro name symbol.
+    pub fn current_macro(&self) -> Option<Symbol> {
+        if let ExpnKind::Macro(MacroKind::Bang, current_macro) =
+            self.expn_span.ctxt().outer_expn_data().kind
```

Comment:
Note: Since this is in a hot path I'm wondering about how efficient it is. I don't have ideas for a faster way though.

Reply:
Nice catch. That `outer_expn_data()` is a bit heavy. I can't avoid it, but I can probably limit the number of times it's called to once per `CoverageSpan`, and it looks likely that I may be calling it multiple times per `CoverageSpan` now.

I will make this change. 

As for the implementation, I think this is consistent with best practices for similar methods (for example, https://github.com/rust-lang/rust/blob/bd9556956ade485ae540fa6b25d2d2b1b2e8b958/compiler/rustc_span/src/lib.rs#L526-L533).

Reply:
Thanks for the catch! I've improved this.